lib/screen/exhibition_details/exhibition_detail_bloc.dart

```diff
@@ -18,10 +22,33 @@ class ExhibitionDetailBloc
         _feralFileService.getExhibitionArtworks(event.exhibitionId,
             withSeries: true)
       ]);
-      final exhibitionDetail = ExhibitionDetail(
-          exhibition: result[0] as Exhibition,
-          artworks: result[1] as List<Artwork>);
+      final exhibition = result[0] as Exhibition;
+      final artworks = result[1] as FeralFileListResponse<Artwork>;
+      final exhibitionDetail =
+          ExhibitionDetail(exhibition: exhibition, artworks: artworks.result);
+      emit(state.copyWith(exhibitionDetail: exhibitionDetail));
+
+      if (artworks.paging.shouldLoadMore) {
+        add(LoadMoreArtworkEvent(artworks.paging.limit, _limit));
+      }
+    });
+
+    on<LoadMoreArtworkEvent>((event, emit) async {
+      log.info(
+          'LoadMoreArtworkEvent: offset=${event.offset}, limit=${event.limit}');
+      final result = await _feralFileService.getExhibitionArtworks(
+          state.exhibitionDetail!.exhibition.id,
+          withSeries: true,
+          offset: event.offset,
+          limit: event.limit);
+      final exhibitionDetail = state.exhibitionDetail!.copyWith(
+        artworks: state.exhibitionDetail!.artworks! + result.result,
+      );
       emit(state.copyWith(exhibitionDetail: exhibitionDetail));
+
+      if (result.paging.shouldLoadMore) {
+        add(LoadMoreArtworkEvent(event.offset + event.limit, event.limit));
```

Comment:
What is the purpose of pagination here? I see it still get all exhibition artworks, just calling multiple APIs continuously instead of calling 1 API to get all data. 
It makes sense Series detail page (call api to load more data when user scroll to the bottom)

I think we should improve the data fetching in this page by using getExhibitionSeries(includeFirstArtwork) API, don't use getExhibitionArtworks anymore.

Reply:
Feral file server limit number of object return to 300.
We need to query all artworks because it was in our requirement to be able cast all exhibition artwork.
If it not the case now, we should disable loading second page, move on and make a proper fix for this

Reply:
Yes, I got it. Please change to use the API getExhibitionSeries instead, we don't need to query all artworks, it doesn't make sense.

Reply:
changing from ExhibitionDetail to Exhibition required some effort to implement and test, and it's out of this scope, so we need another ticket to do that